corehq/apps/userreports/reports/view.py

```diff
@@ -392,6 +403,12 @@ def _get_context_for_saved_report(report_config):
             'datespan_filters': ReportConfig.datespan_filter_choices(self.datespan_filters, self.lang),
         }
 
+        if saved_report_config:
+            # override filter_context with the saved report config filters
+            context.update({'filter_context': self.filter_context_for_saved_report(saved_report_config)})
```

Comment:
Do you know if this is actually necessary?  Know what `default_config` is for if not rendering the filters?  Maybe pre-populating the "save report" modal?

Reply:
Yeah I should have specified this up above because `default_config` is used client-side to update saved report filters right now, but it doesn't work. There are a couple of reasons I went with this approach instead of using `default_config` to update the DOM client side. 
1) There's a specific format for filter values (`{"value": <value>,  "display": <display>}`) which the client side didn't know about (unless this standard across HQ in which case this point is moot. 
2) The changes needed client side were starting to feel unnecessarily complex. Right now this code exists but does not work:
https://github.com/dimagi/commcare-hq/blob/1b34ad8f49a281e5bdf146127bf7a544dc4245bd/corehq/apps/reports/static/reports/js/report_config_models.js#L140-L146
We would need to change `$element.val()` to `$element.attr('data-initial-values', <correctly formatted dict>)`, keeping in mind that format is not something client side is aware of, and would have to add another case for Datespan filters since they have a different attribute to update/different format to be aware of.